service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java

```diff
@@ -154,9 +154,9 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(@RequestBod
 
     @PostMapping("/submitted")
     public void handleSubmittedEvent(@RequestBody @NotNull CallbackRequest callbackRequest) {
-        applicationEventPublisher.publishEvent(new SubmittedCaseEvent(callbackRequest));
-
-        if (isInReturnedState(callbackRequest.getCaseDetailsBefore())) {
+        if (isInOpenState(callbackRequest.getCaseDetailsBefore())) {
+            applicationEventPublisher.publishEvent(new SubmittedCaseEvent(callbackRequest));
+        } else if (isInReturnedState(callbackRequest.getCaseDetailsBefore())) {
```

Comment:
Stupid question, for the future - should. we throw illegalStateException in. else block? If someone adds a new state, it will fail silently, and with the exception it might be easier to find, thoughts?

Reply:
It wont fail, it just wont do anything, which i think is fine as it it not clear what would be business req for this future story